### PR TITLE
Fixes #126 - Add seo page to style guides

### LINKED ISSUE #126: Add SEO section for UXD

### PR DESCRIPTION
**Pull request by goodwinchris**

### CODE REVIEW

File: ux-dev/seo.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+As search engine algorithms have evolved over time, they’ve begun to strongly favor sites that simply have well-structured, relevant, and engaging content.
+
+Search engine rankings are speculated to give significant weight to sites with:
+* Semantic markup
+* Responsive web design
+* Strong adherence to accessibility standards
+
+Positive, organic user engagement metrics (low bounce rate, time on site, number of pages per session, etc.) also support better page rank as an indicator of good content quality and site credibility.
+
+*Don’t* put too much effort into chasing “advanced” SEO strategies.
```
Comment (danrichman) [inline ux-dev/seo.md]: I'm not 100% sure what this means. 🤔 I think it means to keep it simple?

File: ux-dev/seo.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+Sites that place too much effort on building backlinks and other tactic-driven strategies are sacrificing resources that instead could be directed at improving the user experience and content.
```
Comment (danrichman) [inline ux-dev/seo.md]: Not sure that's entirely true. Generally speaking, the overwhelming majority of SEO is determined by factors that are out of the control of web developers—which are mainly meaningful links from external sites, back to the content. The original "PageRank" algorithm was almost entirely based on this metric and is widely seen as the most important ranking signal.

Obviously this comes organically, if one builds a great website. However, it seems like it's our responsibility to at least inform clients that the # 1 thing they can do to improve their SEO is to get other sites to link to their content—whether it be through social media or dedicated arrangements. If we don't at least communicate this reality to clients, I don't think any SEO tricks will budge their rankings much. I feel like they need to be aware of that.

File: ux-dev/seo.md [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+Many of these efforts produce diminishing returns on time invested, and most meaningful metrics such as site reputation can be improved organically with strong UX and engaging content.
```
Comment (danrichman) [inline ux-dev/seo.md]: I 100% agree with this. But I feel like we could clarify. Site reputation is mostly measured by _quality_ links back to a client's content. We should at least inform clients that it's up to them to strive for this highly significant metric of SEO—even if they simply focus on that organically, as you say.

File: ux-dev/seo.md [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+* Target keyword phrase is optional if character count allows
+* Meta description (get as close to *155 characters* as possible, but try not to go over)
+* Meta description should repeat same *page name*, *target keyword phrase*, and *brand name* in well-crafted narrative copy that *accurately describes the page content* and encourages users to click the link to learn more or take action.
+* *Don’t* add meta keywords
+  * They are not considered necessary and even *considered detrimental by some SEO experts*. They were historically “stuffed” by users who were trying to inorganically get better search rankings by jamming in tons of irrelevant keywords. At best, most search engines don’t give any weight to this content (*most search engines stopped using them for web ranking in 2009*), and some may view it as less credible content.
+  * Related article: https://yoast.com/meta-keywords/
+STUB Need information on social media meta content and sharing post/preview images
+
+## Page related
+* Have a *single*, unique `h1` on every page
```
Comment (danrichman) [inline ux-dev/seo.md]: I believe that, ideally, `h1` should also be the first header tag to appear in the markup structure. In other words, it's not considered to be hierarchically correct to have an `h2` or `h3` appear in the markup before an `h1`.

File: ux-dev/seo.md [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+* Ensure all images have relevant, descriptive alt text
```
Comment (danrichman) [inline ux-dev/seo.md]: There is some nuance on this topic. 

When an image has no function, because they are not a link and are not clickable, it is considered best-practices—from an AX standpoint—to have empty `alt` attributes—so as not to add redundant or extraneous text for the user.

See explanation: https://webaim.org/techniques/alttext/

>If the content that the image conveys is presented within text in the surrounding context of the image, then an empty alt attribute may suffice.
>
>...
>
>Alternative text _should not_:
>• be redundant (be the same as adjacent or body text).
>• use the phrases "image of…" or "graphic of…".

If the image serves no specific function (as a link) to the screen reader user and is not specifically described or referred to in the surrounding text (i.e. "See Figure 1"), it is probably better to have an empty `alt` attribute.

File: ux-dev/seo.md [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+* Use relevant microdata where appropriate to help search engines better understand the meaning of important content
+  * Related article: http://schema.org/docs/gs.html
```
Comment (danrichman) [inline ux-dev/seo.md]: This is actually a big deal. I feel like schema.org maybe isn't given enough emphasis in these guidelines. Structuring data is a _significant_ way to make sure data is appropriately indexed. That in conjunction with site maps can be a big deal for indexing—particularly when sites can be javascript-heavy.

I would argue that schema.org is the equivalent of AX for search bots. 😄

Comment (danrichman) [reply ux-dev/seo.md]: I really like the idea of stressing the idea of structuring data, via schema.org though. I feel like all of our clients' data should be structured in that manner. Not only does that hand deliver the content to search engines, but it's high quality billable hours that every client should consider investing in

Using the (schema.org) structured data checker is a great tool and a great way to illustrate how it works.

https://search.google.com/structured-data/testing-tool

Example [test](https://search.google.com/structured-data/testing-tool#url=https%3A%2F%2Fitunes.apple.com%2Fus%2Falbum%2F1989%2Fid907242701)

**Comment by ChrisGoodwin** [comment]
Looks like italics are filling where bold should be. NOT ready for merge.

File: ux-dev/seo.md [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+
+Search engine rankings are speculated to give significant weight to sites with:
+* Semantic markup
+* Responsive web design
+* Strong adherence to accessibility standards
+* Keep the page lightweight and mobile performant
+
+Positive, organic user engagement metrics (low bounce rate, time on site, number of pages per session, etc.) also support better page rank as an indicator of good content quality and site credibility.
+
+**Don’t** put too much effort into chasing “advanced” SEO strategies, we can spend our time better buiding a better site which will also help SEO. "Keep it simple"
```
Comment (davidluhr) [inline ux-dev/seo.md]: "better" is redundant. Maybe rephrase to "we can instead spend our time building a better site…"
